lib/route.js

```diff
@@ -261,27 +255,21 @@ function buildRouting (options) {
         if (opts.schema) {
           context.schema = normalizeSchema(context.schema)
 
-          const schemaBucket = this[kSchemas]
-          if (!opts.validatorCompiler && (
-            !this[kValidatorCompiler] ||
-            (this[kValidatorCompiler] && schemaBucket.hasNewSchemas()))) {
-            // if the instance doesn't have a validator, build the default one for the single fastify instance
-            this.setValidatorCompiler(buildPerformanceValidator(schemaBucket.getSchemas(), this[kOptions].ajv))
+          const schemaController = this[kSchemaController]
+          if (!opts.validatorCompiler) {
+            schemaController.setupValidator(this[kOptions])
```

Comment:
I'm propagating the server option to be framework agnostic: would you like instead to define a generic alias per `ajv` option?

Reply:
I don't understand the question.

Reply:
Right now there is the `ajv` option that is used to setup the default ajv instance in fastify.
Moving the JSON schema's logic in the `schemaController` (that it is ajv by default, but users can implement the `joi` version) it seems odd to me to set as input the `options.ajv` field.

For this reason, I have provided the whole server option object to the `setupValidator` and `setupSerializer`.